syntax = "proto3";

// For use with creation_time
import "google/protobuf/timestamp.proto";

message LocationMessageRequest {
    int32 person_id = 1;
    google.protobuf.Timestamp creation_time = 2;
    float latitude = 3;
    float longitude = 4;
}

message RetrieveMessageRequest {
    int32 id = 1;
}

message LocationMessageResponse {
    int32 id = 1;
    int32 person_id = 2;
    string coordinate = 3;
    string creation_time = 4;
}

service LocationService {
    rpc Get(RetrieveMessageRequest) returns (LocationMessageResponse);
    rpc Create(LocationMessageRequest) returns (LocationMessageResponse);
}